### DISCUSSION TITLE
Need assistance counting button presses using IRQ Rising

### DISCUSSION THREAD

**Discussion by bfpa40** [micropython/micropython#11575] [RP2040 / Pico]
My question deals with IRQ's and button presses on Miccropython. I am using Micropython on Thonny and a RaspberryPi Pico. I have pin14 setup to react to a rising IRQ upon which time it prints "button pushed". Now how would I have those rising IRQ's caused by a button push increment to a given count and when that count is reached make pin13 high to light an LED? I have pin14 setup and its working I have pin13 setup with a led. But i am new and trying to tie together an IRQ and a Count to make pin13 go high is eluding me. Coffee isnt helping.... Following is what I currently have... 
```
`
import machine
import utime

button = machine.Pin(14, machine.Pin.IN, machine.Pin.PULL_DOWN)
led = machine.Pin(13, machine.Pin.OUT)

count = 0

def button_handler(pin):
    utime.sleep_ms(100)
    if pin.value():
        print("button pushed")
        print(count)
        led.toggle()
        
button.irq(trigger=machine.Pin.IRQ_RISING, handler=button_handler)`
```

-Brian-

**Comment by 2dof** [comment]
You forgot increment your counter ....
 ``` def button_handler(pin):
           global count
           count+=1
             ...
```

Go to [Raspberry Pi Pico Interrupts & Button Interfacing Tutorial Using MicroPython](https://electrocredible.com/raspberry-pi-pico-external-interrupts-button-micropython)  - good explanation and examples .
also https://www.upesy.com/blogs/tutorials/micropython-raspberry-pi-pico-gpio-pins-usage  is good tutorial.

**Comment by bfpa40** [comment]
Excellent I had added the global to the count but the articles you sent put me on the right track and I now have it working!  Thank You!

-Brian-
